### DISCUSSION TITLE
Block sealing failed: unauthorized

### DISCUSSION THREAD

**Discussion by deepalt92** [ConsenSys/quorum#1451] [Q&A] [answered]
Has anyone come across a block sealing failed warning (see screenshot below)? I do not get receipts of the transactions I send to quorum so I am assuming it's a sealing issue.
![image](https://user-images.githubusercontent.com/10164213/179286981-3ae58a6d-d398-495d-a0b9-1397573fd3b8.png)
I used quorum with the latest commit "commit 12fe34183274ef8f36560b6f00fd0ecc6b984b9d"  and compiled it on go1.16.2 . Then I followed the steps here: https://consensys.net/docs/goquorum/en/latest/tutorials/private-network/create-qbft-network/) to build a  quorum QBFT private network.

I ran 4 validators.  I tried to start a quorum node as mentioned in the docs:
```
export ADDRESS=$(grep -o '"address": *"[^"]*"' /home/deepal/QBFT-Network/Node-0/data/keystore/accountKeystore | grep -o '"[^"]*"$' | sed 's/"//g')
export PRIVATE_CONFIG=ignore
/home/deepal/go/src/quorum/build/bin/geth --datadir data init /home/deepal/QBFT-Network/Node-0/data/genesis.json
/home/deepal/go/src/quorum/build/bin/geth --datadir data \
    --networkid 161 --nodiscover --verbosity 5 \
    --syncmode full \
    --istanbul.blockperiod 5 --mine --miner.threads 1 --miner.gasprice 0 --emitcheckpoints \
    --http --http.addr 127.0.0.1 --http.port 22001 --http.corsdomain "*" --http.vhosts "*" \
    --ws --ws.addr 127.0.0.1 --ws.port 8546 --ws.origins "*" \
    --http.api admin,eth,debug,miner,net,txpool,personal,web3,istanbul \
    --ws.api admin,eth,debug,miner,net,txpool,personal,web3,istanbul \
    --port 30303 \
    --unlock ${ADDRESS} --allow-insecure-unlock --password /home/deepal/QBFT-Network/Node-0/data/keystore/accountPassword \
```
But, I got this  error (although I had an accountPassword and accountKeyStore file in the specific path):

> Fatal: Failed to unlock account 167cafe7b187e1d7e37a89daaff9904177e83579 (no key for given address or file)

Then I decided to comment out  

> --unlock ${ADDRESS} --allow-insecure-unlock --password /home/deepal/QBFT-Network/Node-0/data/keystore/accountPassword \

and used the following command to start each quorum node:
```
export ADDRESS=$(grep -o '"address": *"[^"]*"' /home/deepal/QBFT-Network/Node-0/data/keystore/accountKeystore | grep -o '"[^"]*"$' | sed 's/"//g')
export PRIVATE_CONFIG=ignore
/home/deepal/go/src/quorum/build/bin/geth --datadir data init /home/deepal/QBFT-Network/Node-0/data/genesis.json
/home/deepal/go/src/quorum/build/bin/geth --datadir data \
    --networkid 161 --nodiscover --verbosity 5 \
    --syncmode full \
    --istanbul.blockperiod 5 --mine --miner.threads 1 --miner.gasprice 0 --emitcheckpoints \
    --http --http.addr 127.0.0.1 --http.port 22001 --http.corsdomain "*" --http.vhosts "*" \
    --ws --ws.addr 127.0.0.1 --ws.port 8546 --ws.origins "*" \
    --http.api admin,eth,debug,miner,net,txpool,personal,web3,istanbul \
    --ws.api admin,eth,debug,miner,net,txpool,personal,web3,istanbul \
    --port 30303
```
Then, I was able to start the nodes without issue. But when sending  transactions, I got the error:
> Block sealing failed:unauthorized
 
Thank you :)
Looking forward to hearing from someone.

**Comment by antonydenyer** [comment, marked as answer]
Make sure your keys directory contains an accountKeystore file that contains the key for the account you're trying to unlock. 

Also make sure that your validators are the ones specified in extradata or sue smart contract validators. 

https://consensys.net/docs/goquorum/en/latest/configure-and-manage/configure/consensus-protocols/qbft/?h=extradata#add-and-remove-validators

https://consensys.net/docs/goquorum/en/latest/tutorials/private-network/create-qbft-network/
